### PR TITLE
Assigment 4

### PR DESCRIPTION
**Pull request by AndreaMejiaCh**
tarea 6 y 7

### CODE REVIEW

File: Assignment4/andreamejia-22c063/task06_AndreaMejia.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#g.add((ns.Researcher,RDF.type,Literal("Jane Smith")))
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+# TO DO
+# Visualize the results
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+janeSmithURI = g.value(None,RDF.type,ns.Researcher)
+g.add((janeSmithURI,vcard.FN,Literal("Jane Smith")))
+g.add((janeSmithURI,vcard.GIVEN,Literal("Jane")))
```
Comment (dgarijo) [inline Assignment4/andreamejia-22c063/task06_AndreaMejia.py:66]: the correct prop is Given (check it in vcard)

File: Assignment4/andreamejia-22c063/task07_AndreaMejía.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+from types import prepare_class
+from rdflib.plugins.sparql import prepareQuery
+from rdflib.namespace import RDF, RDFS
+ns = Namespace("http://somewhere#")
+# TO DO
+# Visualize the results
+q1 =prepareQuery('''
+  SELECT ?Subject WHERE {
+      ?Subject rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/andreamejia-22c063/task07_AndreaMejía.py:34]: Does not retrieve the subclasses of subclasses of living thing

File: Assignment4/andreamejia-22c063/task07_AndreaMejía.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+
+q2 = prepareQuery('''
+  SELECT ?Subject ?Subject2 WHERE
+  {
+
+    {?Subject rdf:type ns:Person}
+    union
+    {
+      ?S rdfs:subClassOf ns:Person.
```
Comment (dgarijo) [inline Assignment4/andreamejia-22c063/task07_AndreaMejía.py:57]: same issue as above. Use `*`

File: Assignment4/andreamejia-22c063/task07_AndreaMejía.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+for r in g.query(q3):
+   print(r.Subject,r.p)
+
+"""**TASK 7.4:  List the name of the persons who know Rocky**"""
+
+FOAF = Namespace("http://xmlns.com/foaf/0.1/")
+q4 = prepareQuery('''
+  SELECT ?conoce WHERE
+  {
+    ns:RockySmith foaf:knows ?conoce.
```
Comment (dgarijo) [inline Assignment4/andreamejia-22c063/task07_AndreaMejía.py:103]: You are not returning what the exercise is asking for.

File: Assignment4/andreamejia-22c063/task07_AndreaMejía.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+for r in g.query(q4):
+   print(r.conoce)
+
+"""**Task 7.5: List the entities who know at least two other entities in the graph**"""
+
+q5 = prepareQuery('''
+  SELECT ?subject ?o WHERE
+  {
+    ?subject foaf:knows ?o.
```
Comment (dgarijo) [inline Assignment4/andreamejia-22c063/task07_AndreaMejía.py:117]: You are not returning what the exercise is asking for.

Also, no exercises are in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address pending changes

**Comment by anaigmo** [comment]
Cierro PR, correcciones en https://github.com/FacultadInformatica-LinkedData/Curso2023-2024-DataScience/pull/97

**Pull request closed without merging**